File: 05-variable-scopes/expressions/DivExpression.cpp
#include "DivExpression.h"

DivExpression::DivExpression(
    Expression *e1, Expression *e2
): first(e1), second(e2) {}

int DivExpression::eval() const {
    return first->eval() / second->eval();
}

void DivExpression::Accept(Visitor* visitor) {
    visitor->Visit(this);
}

File: 2023-fall/nodes/expressions/DivExpression.py
from nodes.expressions.Expression import Expression


class DivExpression(Expression):
    def __init__(
            self,
            left: Expression,
            right: Expression
        ) -> None:
        self.left = left
        self.right = right

    def accept(self, visitor):
        return visitor.visit_add_expression(self)
